NitroxServer/Server.cs

```diff
@@ -44,6 +46,51 @@ public Server(ServerConfigReader configReader)
 
         public void Start()
         {
+            NetworkInterface[] allInterfaces = NetworkInterface.GetAllNetworkInterfaces();
+            foreach(NetworkInterface eachInterface in allInterfaces)
+            {
+                if (eachInterface.Name == "Hamachi")
+                {
+                    string hamachiAddresses = "";
+                    int count = 0; //Multiple Hamachi IP's , IPv4 and IPv6
+                    foreach(IPAddressInformation hamachiIP in eachInterface.GetIPProperties().UnicastAddresses)
+                    {
+                        if (!hamachiIP.Address.ToString().Contains("fe80::"))
+                        {
+                            if (count > 0)
+                            {
+                                hamachiAddresses = hamachiAddresses + " or " + hamachiIP.Address.ToString();
+                            }
+                            else
+                            {
+                                hamachiAddresses = hamachiIP.Address.ToString();
+                            }
+                        }
+                        count += 1;
+                    }
+                    Log.Info("If using Hamachi, use the following IP: " + hamachiAddresses);
+                }
+                if (!(eachInterface.GetIPProperties().GatewayAddresses.Count==0)) // To avoid VMWare / other virtual interfaces
+                {
+                    foreach (IPAddressInformation eachIP in eachInterface.GetIPProperties().UnicastAddresses)
+                    {
+                        char splitter = '.';
+                        string[] splitIpParts = eachIP.Address.ToString().Split(splitter);
+                        int secondPart = 0;
+                        if (splitIpParts.Length>1)
+                        {
+                            int.TryParse(splitIpParts[1], out secondPart);
+                        }
+                        if (splitIpParts[0] == "10" || (splitIpParts[0] == "192" && splitIpParts[1] == "168") || (splitIpParts[0] == "172" && (secondPart > 15 && secondPart < 32))) //To get if IP is private
```

Comment:
Since this is the crux of this foreach maybe put it higher in this foreach and use `continue;` to skip if it doesn't meet these checks. Early exit > nested ifs.

Reply:
As in the if statement, it requires an integer (secondPart), which is parsed above the if statement. As this integer is only used once, do I just remove the parsing code above the if statement and parse the integer in the actual if statement instead?

Reply:
Hmm.. I tried cleaning it up but this "determine private ip address" doesn't seem to be much smaller than this. You can ignore this conversation.